lib/ts/recipe/session/sessionAuth.tsx

```diff
@@ -161,8 +160,42 @@ const SessionAuth: React.FC<PropsWithChildren<SessionAuthProps>> = ({ children,
         }
     }, []);
 
+    const getRedirectPathFromInvalidClaimsOrRedirect = async (claims: ClaimValidationError[]) => {
```

Comment:
This should use `useCallback` and be added to the dependency arrays where called.